desktop/src/main/java/bisq/desktop/util/validation/PhoneNumberValidator.java

```diff
@@ -0,0 +1,143 @@
+package bisq.desktop.util.validation;
+
+import bisq.core.locale.Res;
+import bisq.core.util.validation.InputValidator;
+
+import lombok.Getter;
+
+import javax.annotation.Nullable;
+
+/**
+ * Performs lenient validation of international phone numbers, and transforms given
+ * input numbers into E.164 international form.  The E.164 normalized phone number
+ * can be accessed via {@link #getNormalizedPhoneNumber} after successful
+ * validation -- {@link #getNormalizedPhoneNumber} will return null if validation
+ * fails.
+ * <p>
+ * Area codes and mobile provider codes are not validated, but all numbers following
+ * calling codes are included in the normalized number.
+ * <p>
+ * @see  bisq.desktop.util.validation.CountryCallingCodes
+ */
+public class PhoneNumberValidator extends InputValidator {
+    /**
+     * ISO 3166-1 alpha-2 country code
+     */
+    private final String isoCountryCode;
+    /**
+     * Immutable map of international calling codes.
+     */
+    private final CountryCallingCodes countryCallingCodes;
+    /**
+     * The international calling code mapped to the 'isoCountryCode' constructor argument.
+     */
+    @Nullable
+    @Getter
+    private final String callingCode;
+    /**
+     * The normalized (digits only) representation of an international calling code.
+     */
+    private final String normalizedCallingCode;
+    /**
+     * Phone number in E.164 format.
+     */
+    @Nullable
+    @Getter
+    private String normalizedPhoneNumber;
+
+    // Hide no-arg constructor
+    private PhoneNumberValidator() {
+        this.isoCountryCode = null;
+        this.countryCallingCodes = null;
+        this.callingCode = null;
+        this.normalizedCallingCode = null;
+    }
+
+    public PhoneNumberValidator(String isoCountryCode) {
+        this.isoCountryCode = isoCountryCode;
+        this.countryCallingCodes = new CountryCallingCodes();
+        this.callingCode = countryCallingCodes.getCallingCode(isoCountryCode);
+        this.normalizedCallingCode = countryCallingCodes.getNormalizedCallingCode(isoCountryCode);
+    }
+
+
+    @Override
+    public ValidationResult validate(String input) {
+        normalizedPhoneNumber = null;
+        ValidationResult result = super.validate(input);
+        if (!result.isValid) {
+            return result;
+        }
+        final String trimmedInput = input.trim();
+        boolean isCountryDialingCodeExplicit = trimmedInput.startsWith("+");
+        // remove non-digit chars:  plus sign, dashes, parens, hashes, slashes and white spaces
+        String pureNumber = input.replaceAll("[\\+\\s+()\\-\\−\\#\\\\]", "");
```

Comment:
Instead of removing specific chars remove everything that is not a digit: `^\\d` (regexp not tested).

Reply:
Letters need to be included in the pureNumber string for the isPositiveNumber test.  I had a US number with letters, and I'm sure they still exist in other countries.  A number with letters may be valid, but the user will be prompted to convert them to  numbers for successful validation.

Reply:
Sure but that does not change the essence. Remove everything that is not digit-or-letter. That is better approach than haunting for specific chars. Try `"[^A-Za-z0-9]"` and adjust regarding expected capitalization.

Reply:
Yeah, "[^A-Za-z0-9]" is a much better regex for this purpose.  I'll make that change soon, but maybe not tonight.